plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1236,39 +1234,55 @@ public void executeRemoveOrphanFiles(ConnectorSession session, IcebergTableExecu
 
         long expireTimestampMillis = session.getStart().toEpochMilli() - retention.toMillis();
         removeOrphanFiles(table, session, executeHandle.getSchemaTableName(), expireTimestampMillis);
-        removeOrphanMetadataFiles(table, session, executeHandle.getSchemaTableName(), expireTimestampMillis);
     }
 
     private void removeOrphanFiles(Table table, ConnectorSession session, SchemaTableName schemaTableName, long expireTimestamp)
     {
-        Set<String> validDataFilePaths = stream(table.snapshots())
-                .map(Snapshot::snapshotId)
-                .flatMap(snapshotId -> stream(table.newScan().useSnapshot(snapshotId).planFiles()))
-                .map(fileScanTask -> fileName(fileScanTask.file().path().toString()))
-                .collect(toImmutableSet());
-        Set<String> validDeleteFilePaths = stream(table.snapshots())
-                .map(Snapshot::snapshotId)
-                .flatMap(snapshotId -> stream(table.newScan().useSnapshot(snapshotId).planFiles()))
-                .flatMap(fileScanTask -> fileScanTask.deletes().stream().map(file -> fileName(file.path().toString())))
-                .collect(Collectors.toUnmodifiableSet());
-        scanAndDeleteInvalidFiles(table, session, schemaTableName, expireTimestamp, union(validDataFilePaths, validDeleteFilePaths), "/data");
+        Set<String> processedManifestFilePaths = new HashSet<>();
+        // Using file names may result in retaining files which could be removed. However, in practice Iceberg metadata
+        // and data files have UUIDs in their names which makes this unlikely. Collecting file names here rather than paths
+        // makes this comparison easier when checking if files should be retained.
```

Comment:
Thanks for adding this explanation.
I don't quiet get it yet. Intuitively, it's as easy to compare paths as it's to compare file names.
What problem did you face where you tried that?

Reply:
`testCleaningUpWithTableWithSpecifiedLocation` was failing because `/a/slash.txt` vs `/a//slash.txt` resolve to the same file but have different paths.

I'll update the comment to address this

Reply:
> because `/a/slash.txt` vs `/a//slash.txt` resolve to the same file but have different paths.

But only one is "correct".

I think i understand. The metadata may contain entries with double slashes. If we're running on a file system that normalizes repeated slashes (like .. not S3), the table works. To avoid breaking it here, we cannot compare paths.